### PR TITLE
mrdox has a rng schema & test to verify.

### PR DESCRIPTION
**Pull request by klemens-morgenstern**

### CODE REVIEW

**Comment by klemens-morgenstern** [comment]
Clang is just missing an install libxml2, so CI setup.

File: mrdox.rnc [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    }*
+
+    SymbolInfo = (
+        Info,
+        Location *
+    )
+
+    TemplateParamInfo = element tparam { attribute decl { text } }
+
+    BaseRecordinfo = element base {
```
Comment (vinniefalco) [inline mrdox.rnc:40]: capital Info?

File: mrdox.rnc [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            NamespaceInfo |
+            RecordInfo    |
+            FunctionInfo  |
+            TypedefInfo   |
+            EnumInfo      |
+            VariableInfo)*
+
+    # JavaDoc not in a nested grammar , so we can convert it to a XSD
+
+    Javadoc = element doc {Brief ?, Node* }
```
Comment (vinniefalco) [inline mrdox.rnc:155]:
```
{ Brief?, Node*, Param*, TParam* }
```

File: mrdox.dtd [hunk 3]
```diff
@@ -1,5 +0,0 @@
-<!ELEMENT mrdox (all, namespace)>
-<!ELEMENT all (symbol)>
-<!ATTLIST symbol name usr #REQUIRED>
-<!ELEMENT namespace>
```
Comment (vinniefalco) [inline mrdox.dtd:4]: LOL

**Comment by vinniefalco** [comment]
The only little wrinkle is I was thinking to remove the dtd and prolog from the emitted files to make them smaller but I guess that is not necessary

**Comment by klemens-morgenstern** [comment]
The reason I added the rnc is so that my IDE (CLion) picks that up. It's not really necessary

File: mrdox.rnc (hunk 2 above)
Comment (klemens-morgenstern) [reply mrdox.rnc:155]: param Tparam are part of node

**Comment by vinniefalco** [comment]
Merged thanks

**Pull request closed without merging**